Rakefile

```diff
@@ -184,562 +184,577 @@ task :'release:docs' => :docs do
   sh "aws s3 cp --recursive doc/ s3://#{S3_BUCKET}/#{S3_DIR}/docs/"
 end
 
+# Declare a command for execution.
+# Jobs are parallelized if running in CI.
+def declare(*args)
+  total_executors = ENV.key?('CIRCLE_NODE_TOTAL') ? ENV['CIRCLE_NODE_TOTAL'].to_i : nil
+  current_executor = ENV.key?('CIRCLE_NODE_INDEX') ? ENV['CIRCLE_NODE_INDEX'].to_i : nil
+
+  if total_executors && current_executor && total_executors > 1
+    @execution_count ||= 0
+    @execution_count += 1
+    sh(*args) if @execution_count % total_executors == current_executor
```

Comment:
What's the logic here exactly? I'm not sure I follow

Reply:
@ericmustin This method splits tests based on the total number of test executors.

If we have 30 Ruby tests total, 10 total executors (CI containers), and we are currently running on executor number 3, for example, `bundle exec rake ci` should only run tests where `test_num % 10 == 3)` which would translate to tests number 3, 13, and 23.